proxy/src/main.rs

```diff
@@ -13,5 +19,7 @@ fn main() {
             process::exit(64)
         }
     };
-    conduit_proxy::Main::new(config, conduit_proxy::SoOriginalDst).run();
+    let main = conduit_proxy::Main::new(config, conduit_proxy::SoOriginalDst);
```

Comment:
Let's add a comment explaining that we're doing this because we'll often be running in a container as PID 1, beyond our control, and the proxy can't always avoid running as PID 1 because of https://github.com/kubernetes/kubernetes/pull/51634.